src/readers/morphologyASC.cpp

```diff
@@ -140,9 +149,23 @@ class NeurolucidaParser
             nb_.addMarker(marker);
             return_id = -1;
         } else if (header.token == Token::CELLBODY) {
-            if (!nb_.soma()->points().empty())
-                throw SomaError(err_.ERROR_SOMA_ALREADY_DEFINED(lex_.line_num()));
-            nb_.soma()->properties() = properties;
+            const auto& soma = nb_.soma();
```

Comment:
I can't say for sure. This type of soma is still a contour. MorphIO does not calculate `surface` or `volume` for contour somas. All other properties are not affected by stack. That hesitates me to introduce a new soma type. What for? MorphIO offers access to points and morphology, that's it. I would actually drop `surface` and `volume` properties as MorphIO does not calculate them for all cases.

Reply:
> That hesitates me to introduce a new soma type. What for?

For one thing, so that a morph read from ASC can be written back to ASC w/ the same structure.

The other problem that I can forsee is the contour might be really strange: if the orientation of the z slices are rotated 180 to each other, then the 'contour' points would cross through the soma.

Reply:
ok, it makes sense to create a new soma type. I will follow the approach of `morphologySWC.cpp` but in general the current soma type detection has a problem #314.

Reply:
Hold off for now; I'm not sure there's a clean solution, and I think we need to ask around first.